### DISCUSSION TITLE
Using Lottie in Avalonia

### DISCUSSION THREAD

**Discussion by OmidID** [AvaloniaUI/Avalonia#5946] [General]
Recently an opensource Lottie project was released base on WPF + SharpDX and i want to know how we can use it in Avalonia too https://github.com/ascora/LottieSharp
I want to know how we can implement D2dControl for Avalonia 
https://github.com/ascora/LottieSharp/blob/master/LottieSharp/WpfSurface/D2dControl.cs

**Comment by Gillibald** [comment]
If you stick to the Direct2D1 backend it should be possible to port this to Avalonia. A more decoupled version that supports more backends would make more sense for Avalonia.

**Comment by OmidID** [comment]
@Gillibald It's base on SharpDX now. To make it portable backends it's need to modify lot of things. right?

**Comment by Gillibald** [comment]
If they just use Direct2D1 API it would be easier to port this to Avalonia because we kind of expose the RenderTarget API via DrawingContext. Haven't checked the implementation details so I could be wrong.

**Comment by OmidID** [comment]
@Gillibald Another project was released for SkiaSharp https://github.com/yinyue200/LottieSkiaSharp
Now there is 2 separate project exists one for SharpDx and the other SkiaSharp. Do you think is it possible to make it fully compatible with with Avalonia for all platform?

**Comment by Gillibald** [comment]
Looks like it is possible to just rewrite the rendering logic to support the Avalonia render pipeline. I would use LottieSkiaSharp as a starting point and just replace all the Skia related stuff with our render backend. That way you get support for all backends like D2D1 and Skia. Shouldn't be that complicated.

**Comment by jmacato** [comment]
It would be better if the potential Lottie port for Avalonia be hooked to our RenderTimers, having a separate timer screws up rendering and is less performant

**Comment by FabianGolembiewski-sp** [comment]
Are there any updates regarding lottie support?
Are there alternative formats for UX-/UI-Design exports that can be rendered via AvaloniaUI?

**Comment by jmacato** [comment]
https://github.com/AvaloniaUI/Avalonia.Lottie/ check this out @FG-rgb

**Comment by FabianGolembiewski-sp** [comment]
Ah nice one. Thanks for the quick response ;)

**Comment by jmacato** [comment]
not yet ready for primetime but getting there

**Comment by FabianGolembiewski-sp** [comment]
Ok, I tested it, but seem not to be ready yet. Will give it a try in the future.
I would like to stay informed in case of any updates.
(Actually I only see a green or red background, Platform Windows 10 X64).

**Comment by jmacato** [comment]
we're in the middle of developing it. stay tuned by watching the repo.

**Comment by FabianGolembiewski-sp** [comment]
Big thanks. I am looking forward to it.

**Reply by danwalmsley** [reply]
https://github.com/AvaloniaUI/Avalonia.Lottie/

not ready but can follow along here.

**Comment by BaculisBirota** [comment]
Hi, I decided to try Avalonia.Lottie. In development mode the animation is played, but in runtime, the animation is updated only if you change the size of the window (manually expose the rendering area). What could be the problem?
In XAML editor:
![XAML](https://user-images.githubusercontent.com/73808544/140038203-44bb5af6-5c49-46c4-97fc-9728e6a2505d.gif)
In Runtime:
![Runtime](https://user-images.githubusercontent.com/73808544/140038228-51443542-ffc5-406c-94db-3cc19de7dcb2.gif)
